util/TLSProxy/Message.pm

```diff
@@ -198,14 +198,14 @@ sub get_messages
         print "  [".$record->decrypt_data."]\n";
     } elsif ($record->content_type == TLSProxy::Record::RT_ALERT) {
         my ($alertlev, $alertdesc) = unpack('CC', $record->decrypt_data);
-        #All alerts end the test
-        $end = 1;
         #A CloseNotify from the client indicates we have finished successfully
         #(we assume)
-        if (!$server && $alertlev == AL_LEVEL_WARN
+        if (!$end && !$server && $alertlev == AL_LEVEL_WARN
             && $alertdesc == AL_DESC_CLOSE_NOTIFY) {
             $success = 1;
         }
+        #All alerts end the test
+        $end = 1;
```

Comment:
Yeah, this is why this was a question... The test handshake does a controlled mutual shutdown but s_server apparently doesn't bother then, or the proxy doesn't wait for it. I'm not sure either how to fix it with the current code, then; just leave a comment.

What do you mean by two alerts - the client sends the same alert twice? Two different alerts? (I did find one such case recently and I'd not be surprised if there were more. But it could also be that something in the proxy fwding logic is broken and ends up repeating things.)